applications/OptimizationApplication/custom_utilities/response/mass_response_utils.cpp

```diff
@@ -0,0 +1,271 @@
+                const double ly_derivative = ((NodeIndex == 0) - (NodeIndex == 1)) * (DirectionIndex == 1);
+
+                const double lz = rGeometry[0].Z() - rGeometry[1].Z();
+                const double lz_derivative = ((NodeIndex == 0) - (NodeIndex == 1)) * (DirectionIndex == 2);
+
+                const double length = lx * lx + ly * ly + lz * lz;
+                const double length_derivative = 2 * lx * lx_derivative + 2 * ly * ly_derivative + 2 * lz * lz_derivative;
+
+                return 0.5 * length_derivative / std::sqrt( length );
+            };
+            break;
+        case GeometryData::KratosGeometryType::Kratos_Triangle2D3:
+            volume_derivative_method = [](IndexType NodeIndex, IndexType DirectionIndex,  const GeometryType& rGeometry) {
+                return 2.0 * ElementSizeCalculator<2, 3>::AverageElementSize(rGeometry) * ElementSizeCalculator<2, 3>::AverageElementSizeDerivative(NodeIndex, DirectionIndex, rGeometry);
+            };
+            break;
+        case GeometryData::KratosGeometryType::Kratos_Quadrilateral2D4:
+            volume_derivative_method = [](IndexType NodeIndex, IndexType DirectionIndex,  const GeometryType& rGeometry) {
+                return 2.0 * ElementSizeCalculator<2, 4>::AverageElementSize(rGeometry) * ElementSizeCalculator<2, 4>::AverageElementSizeDerivative(NodeIndex, DirectionIndex, rGeometry);
+            };
+            break;
+        case GeometryData::KratosGeometryType::Kratos_Tetrahedra3D4:
+            volume_derivative_method = [](IndexType NodeIndex, IndexType DirectionIndex,  const GeometryType& rGeometry) {
+                return 3.0 * std::pow(ElementSizeCalculator<3, 4>::AverageElementSize(rGeometry), 2) * ElementSizeCalculator<3, 4>::AverageElementSizeDerivative(NodeIndex, DirectionIndex, rGeometry);
+            };
+            break;
+        case GeometryData::KratosGeometryType::Kratos_Prism3D6:
+            volume_derivative_method = [](IndexType NodeIndex, IndexType DirectionIndex,  const GeometryType& rGeometry) {
+                return 3.0 * std::pow(ElementSizeCalculator<3, 6>::AverageElementSize(rGeometry), 2) * ElementSizeCalculator<3, 6>::AverageElementSizeDerivative(NodeIndex, DirectionIndex, rGeometry);
+            };
+            break;
+        // Following is not consistent with the DomainSize calculation, hence commented out for the time being.
+        // case GeometryData::KratosGeometryType::Kratos_Hexahedra3D8:
+        //     volume_derivative_method = [](IndexType NodeIndex, IndexType DirectionIndex,  const GeometryType& rGeometry) {
+        //         return 3.0 * std::pow(ElementSizeCalculator<3, 8>::AverageElementSize(rGeometry), 2) * ElementSizeCalculator<3, 8>::AverageElementSizeDerivative(NodeIndex, DirectionIndex, rGeometry);
+        //     };
+        //     break;
+        default:
+            KRATOS_ERROR << "Non supported geometry type." << std::endl;
```

Comment:
```suggestion
            KRATOS_ERROR << "Non supported geometry type for mass shape sensitivity calculation (CalculateMassShapeSensitivity())." << std::endl;
```
Make the error more specific